src/Benchmarks/Data/EfDb.cs

```diff
@@ -49,9 +57,10 @@ public Task<World> LoadSingleQueryRow()
             for (var i = 0; i < count; i++)
             {
                 var id = _random.Next(1, 10001);
-                var result = await _dbContext.World.AsTracking().FirstAsync(w => w.Id == id);
+                var result = await _firstWorldQuery(_dbContext, id);
+
+                _dbContext.Attach(result).Property("RandomNumber").CurrentValue = _random.Next(1, 10001);
```

Comment:
In the meanwhile I cannot think of a way to do this that takes advantage of compiled queries and doesn't make the code much uglier

cc @anpete @ajcvickers @mikeharder